src/cargo/sources/registry/index.rs

```diff
@@ -277,10 +278,13 @@ impl<'cfg> RegistryIndex<'cfg> {
     ) -> CargoResult<()> {
         let source_id = self.source_id;
         let name = dep.package_name().as_str();
+        let yanked_whitelist = self.yanked_whitelist.clone();
```

Comment:
This is a pretty hot path for Cargo so it'd be best to avoid the clone here, would it be possible to shared borrow after the call to `self.summaries`?

Reply:
Oh right, another idea would be to perhaps pass it in as an argument? I think that this method is already called with extra arguments from the main registry source anyway